src/SourceFile.js

```diff
@@ -20,14 +20,22 @@ export class SourceFile {
   encodedLiterals = new Map();
   warnings = [];
 
-  constructor(sourceCodeString) {
+  constructor(sourceCodeString, customProbes = [], mergeMode = "append") {
     this.tracer = new VariableTracer()
       .enableDefaultTracing()
       .trace("crypto.createHash", {
         followConsecutiveAssignment: true, moduleName: "crypto"
       });
 
-    this.probesRunner = new ProbeRunner(this);
+    let mergedProbes;
+    if (Array.isArray(customProbes) && customProbes.length > 0) {
```

Comment:
Shouldn't `customProbes` always be an array?
If we feel the need to enforce typing, I think we should throw a `TypeError` if our apis are misused.

Reply:
This is shorter like this:
 ```js
let probes = ProbeRunner.Defaults;
if (Array.isArray(customProbes) && customProbes.length > 0) {
  probes = mergeMode === "replace" ? customProbes : [...probes, ...customProbes];
}
```